sources/openeuler-http.go

```diff
@@ -0,0 +1,122 @@
+package sources
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"net/url"
+	"path/filepath"
+
+	"github.com/lxc/distrobuilder/shared"
+)
+
+type openEuler struct {
+	commonRHEL
+	fileName     string
+	checksumFile string
+}
+
+const (
+	isoFileName = "%s-%s-dvd.iso"
+	shaFileName = "%s-%s-dvd.iso.sha256sum"
+)
+
+func (s *openEuler) Run() error {
+	var err error
+	baseURL := fmt.Sprintf("%s/%s/ISO/%s/", s.definition.Source.URL,
+		s.definition.Image.Release,
+		s.definition.Image.Architecture)
+
+	fpath := s.getTargetDir()
+
+	s.fileName = fmt.Sprintf(isoFileName, s.definition.Image.Name, s.definition.Image.Architecture)
+	s.checksumFile = fmt.Sprintf(shaFileName, s.definition.Image.Name, s.definition.Image.Architecture)
+
+	_, err = url.Parse(baseURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse URL %s", baseURL)
+	}
+	//s.definition.Source.SkipVerification ignored here
```

Comment:
Why is this ignored? If the source is http, we want to verify the source using GPG keys.

Reply:
For now, openeuler do not ship GPG signatures with the checksum file, so we just igonre it.

Reply:
OK. Then just please remove that line.